tests/testthat/test-content.R

```diff
@@ -270,6 +270,44 @@ test_that("get_jobs() using the old and new endpoints returns sensible results",
   expect_equal(jobs_v0$status, c(0L, 0L, NA, NA, NA, NA))
 })
 
+with_mock_api({
+  test_that("get_job_list() returns expected data", {
+    client <- Connect$new(server = "http://connect.example", api_key = "not-a-key")
+    item <- content_item(client, "8f37d6e0")
+    job_list <- get_job_list(item)
+
+    expect_equal(
+      job_list[[1]],
+      list(
+        id = "40793542", ppid = "2611431", pid = "2611448", key = "waaTO7v75I84S1hQ",
+        remote_id = NULL, app_id = "52389", variant_id = "0",
+        bundle_id = "127015", start_time = "2024-12-05T16:22:51Z",
+        end_time = NULL, last_heartbeat_time = "2024-12-05T15:23:41Z",
+        queued_time = NULL, queue_name = NULL, tag = "run_app", exit_code = NULL,
+        status = 0L, hostname = "dogfood01", cluster = NULL, image = NULL,
+        run_as = "rstudio-connect", app_guid = "8f37d6e0", client = client
+      )
+    )
+
+    expect_equal(
+      job_list[[length(job_list)]],
+      list(
+        id = "39368207", ppid = "2434183", pid = "2434200", key = "HbdzgOJrMmMTq6vu",
+        remote_id = NULL, app_id = "52389", variant_id = "0",
+        bundle_id = "127015", start_time = "2024-11-15T17:03:00Z",
+        end_time = "2024-11-15T17:06:23Z", last_heartbeat_time = "2024-11-15T17:06:20Z",
+        queued_time = NULL, queue_name = NULL, tag = "run_app",
+        exit_code = 0L, status = 2L, hostname = "dogfood02",
+        cluster = NULL, image = NULL, run_as = "rstudio-connect",
+        app_guid = "8f37d6e0", client = client
+      )
+    )
+  })
```

Comment:
I wonder if it would be good to test a handful of these properties instead of all of them at once. Given the code that is in `get_job_list`, we probably don't have to worry that there's shuffling of the values there (where given the mocks you get a mix of stuff from 40793542 and 39368207 there. So maybe it's "just" testing the `id`, `client`, and `guid` bits (or which ever are the important bits for jobs+logs)? That would be kind(er) to future-you when these change rather than a big diff of a bunch of things that all changed at once, we have a smaller one.

Reply:
I did as you suggested here.